packages/dds/tree/src/feature-libraries/modular-schema/comparison.ts

```diff
@@ -195,6 +195,11 @@ export function allowsRepoSuperset(
 		}
 	}
 	for (const [key, schema] of original.nodeSchema) {
+		// If a nodeStoredSchema in 'A' does not exist in the 'B', 'B' can no longer be considered a superset of
```

Comment:
This check seems like it's already covered by the call into `allowsTreeSuperset` below. Specifically, the `isNeverTree` check on the superset tree normalizes undefined to never (no schema defined for it, hence no content can ever exist).

The bug that this work item refers to the handling of cases where the root field schema is the implicit undefined case in either of the two repositories being compared. You can trace through the implementation, but in most cases it ends up going to `allowsTreeSchemaIdentifierSuperset`, which loses context of what the set of allowed schemas are.

I need to think on this a bit more, but it seems plausible to me that `allowsRepoSuperset` ends up returning the right thing due to subsequent checks in most of these cases anyway. One case that I don't think it gets right is when two schema repositories which have the same content, except one of them explicitly declares allowed root field types to be all types in their document, and the other one has allowed field types `undefined`. This is documented to be equivalent, but I think the current implementation of `allowsRepoSuperset` won't see it that way due to one of the early exits in `allowsTreeSchemaIdentifierSuperset`.

Reply:
Rather than fix the bug to respect the current documented behavior, I wonder if it it's feasible to remove the possibility of undefined from `TreeTypeSet`. I believe this option is not exposed from the simple-tree layer (i.e. users of SharedTree need to supply a bounded set of types for the root field). Not 100% confident that's true though, it'd be good to check with other ST experts.

Reply:
Hey @CraigMacomber, may I know if there are any work items created related to that? Not sure if we should fix this now or wait for a more detailed plan.